doc/sonic-installer/SONiC_Image_Installation_Verification.md

```diff
@@ -0,0 +1,379 @@
+
+- The option of platform AISC type check skip should be added to sonic-installer help menu.
+  - Use `--skip-platform-check` option instead of `--force` as the later one has been used for image force installation which skips image secure boot check.
+```
+admin@vlab-01:~$ sudo sonic-installer install --help
+Usage: sonic-installer install [OPTIONS] URL
+
+  Install image from local binary or URL
+
+Options:
+  -y, --yes
+  -f, --force, --skip-secure-check
+                                  Force installation of an image of a non-
+                                  secure type than secure running image
+  --skip-platform-check           Force installation of an image of of a different       <======
+                                  ASIC type from the current platform
+  --skip_migration                Do not migrate current configuration to the
+                                  newly installed image
+  --skip-package-migration        Do not migrate current packages to the newly
+                                  installed image
+  --skip-setup-swap               Skip setup temporary SWAP memory used for
+                                  installation
+  --swap-mem-size INTEGER         SWAP memory space size  [default: (1024
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --total-mem-threshold INTEGER   If system total memory is lower than
+                                  threshold, setup SWAP memory  [default:
+                                  (2048 MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --available-mem-threshold INTEGER
+                                  If system available memory is lower than
+                                  threhold, setup SWAP memory  [default: (1200
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --help                          Show this message and exit.
+
+```
+
+# 4 Design and Flow Diagrams
+
+## 4.1 Background
+
+### 4.1.1 Design Attempts
+
+**Trial One: Simply use ASIC as comparison**
+
+In SONiC, the current ASIC info can be accessed from `/etc/sonic/sonic_version.yml`. With that, we can compare the `asic_type` with the to-be-installed image's ASIC and check if they match.
+```
+admin@str-s6000-acs-13:~$ cat /etc/sonic/sonic_version.yml
+...
+asic_type: broadcom                         <======
+```
+Need to mention that, it may prompt false negative message. Assume we have an broadcom switch which has already installed mellanox built image as no ASIC check before. Its `asic_type` will be translated to mellanox. Then it will wrongly report asic doesn't match if we try to install the correct broadcom built image.
+
+In ONiE, that info is not accessible. So we cannot do the ASIC check in ONiE.
```

Comment:
>ONiE [](http://example.com/codeflow?start=3&length=4)

The capitalization is not as official. Please use `ONIE` or `onie`. ref: [https://www.opencompute.org/wiki/Networking/ONIE](https://www.opencompute.org/wiki/Networking/ONIE)